rotkehlchen/tasks/manager.py

```diff
@@ -513,26 +510,13 @@ def _maybe_update_snapshot_balances(self) -> None:
     def _maybe_update_curve_pools(self) -> None:
         """Function that schedules curve pools update task if either there is no curve pools cache
         yet or this cache has expired (i.e. it's been more than a week since last update)."""
-        curve_lp_tokens = GlobalDBHandler().get_general_cache_values(
-            key_parts=[GeneralCacheType.CURVE_LP_TOKENS],
+        self.greenlet_manager.spawn_and_track(
```

Comment:
no this approach is not okay. You are always scheduling it as far as the task manager is concerned. Rebase on develop on top of alexey's work and adjust.

What you should do is either move the timestamp check outside of `self.update_curve_pools_cache` and have it here to check. If it's happened just don't schedule again.

Reply:
About moving it is not possible since the method is called when querying the price of a curve token. I guess I'll have to duplciate the check in the task

Reply:
I meant what you said
> And have it used by curve_protocol_cache_is_queried and also here?

Yes, I can do that.